app/javascript/util/languageUtil.ts

```diff
@@ -0,0 +1,115 @@
+import { t, addTranslation } from "@sf-digital-services/ui-components"
+
+import { cleanPath, getPathWithoutLeadingSlash } from "./urlUtil"
+
+type PhraseBundle = Record<string, unknown>
+export interface LangConfig {
+  prefix: LanguagePrefix
+  isDefault: boolean
+  getLabel: () => string
+  load: () => Promise<PhraseBundle>
+}
+
+export enum LanguagePrefix {
+  English = "en",
+  Spanish = "es",
+  Chinese = "zh",
+  Tagalog = "tl",
+}
+
+export const LANGUAGE_CONFIGS: Record<LanguagePrefix, LangConfig> = {
+  [LanguagePrefix.English]: {
+    prefix: LanguagePrefix.English,
+    isDefault: true,
+    getLabel: () => t("languages.en"),
+    load: async () => await import("../../assets/json/translations/react/en.json"),
+  },
+  [LanguagePrefix.Spanish]: {
+    prefix: LanguagePrefix.Spanish,
+    isDefault: false,
+    getLabel: () => t("languages.es"),
+    load: async () => await import("../../assets/json/translations/react/es.json"),
+  },
+  [LanguagePrefix.Chinese]: {
+    prefix: LanguagePrefix.Chinese,
+    isDefault: false,
+    getLabel: () => t("languages.zh"),
+    load: async () => await import("../../assets/json/translations/react/zh.json"),
+  },
+  [LanguagePrefix.Tagalog]: {
+    prefix: LanguagePrefix.Tagalog,
+    isDefault: false,
+    getLabel: () => t("languages.tl"),
+    load: async () => await import("../../assets/json/translations/react/tl.json"),
+  },
+}
+
+const loadDefaultTranslation = async (): Promise<PhraseBundle> =>
+  await Object.values(LANGUAGE_CONFIGS)
+    .find((config) => config.isDefault)
+    .load()
+
+/**
+ * Load the required translation phrases for the given language prefix.
+ *
+ * Ex: prefix="es" will load Spanish phrases with English as backup
+ *
+ * @param prefix is a string, cannot be blank or null.
+ */
+export const loadTranslations = async (prefix: LanguagePrefix): Promise<void> => {
+  addTranslation(await loadDefaultTranslation())
+
+  const config = LANGUAGE_CONFIGS[prefix]
+
+  if (!config.isDefault) {
+    addTranslation(await config.load())
+  }
+}
+
+export const toLanguagePrefix = (routePrefix: string | undefined): LanguagePrefix => {
+  switch (routePrefix) {
+    case LanguagePrefix.Spanish:
```

Comment:
I know this looks a little weird but it's the only way to get a string enum value